File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/PokemonDetailsContract.java
package com.mdgd.pokemon.ui.pokemon;

import com.mdgd.mvi.fragments.FragmentContract;
import com.mdgd.pokemon.ui.pokemon.items.PokemonProperty;

import java.util.List;

public class PokemonDetailsContract {

    public interface ViewModel extends FragmentContract.ViewModel<View> {
        void setPokemonId(long pokemonId);
    }

    public interface View extends FragmentContract.View {
        void setItems(List<PokemonProperty> items);
    }

    public interface Host extends FragmentContract.Host {
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/PokemonDetailsContract.kt
package com.mdgd.pokemon.ui.pokemon

import com.mdgd.mvi.fragments.FragmentContract
import com.mdgd.pokemon.ui.pokemon.dto.PokemonProperty

class PokemonDetailsContract {
    interface ViewModel : FragmentContract.ViewModel<View> {
        fun setPokemonId(pokemonId: Long)
        fun onBackPressed()
    }

    interface View : FragmentContract.View {
        fun setItems(items: List<PokemonProperty>)
        fun goBack()
    }

    interface Host : FragmentContract.Host {
        fun onBackPressed()
    }
}
